src/lib/state.py

```diff
@@ -148,7 +149,7 @@ def apply_controlled(self, gate, control, target) -> None:
 # be distinguished when multiplied with an arbitrary complex number, aka,
 # global phase.
 #
-def qubit(alpha:float=None, beta:float=None) -> State:
+def qubit(alpha: Optional[float]=None, beta: Optional[float]=None) -> State:
```

Comment:
Now that I'm looking at this... alpha and beta should be complex numbers. Perhaps replace with np.complexfloating?

Reply:
Done.